libs/barista-components/table/src/selection/selectors.ts

```diff
@@ -0,0 +1,184 @@
+import { DtTableSelection } from './selection';
+import { Subject, Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { DtTableDataSource } from '../table-data-source';
+
+/** Handles row selection in combination with the DtTableSelection directive on the table */
+@Component({
+  selector: 'dt-table-row-selector',
+  template: ` <dt-cell class="dt-selectable-cell">
+    <dt-checkbox
+      [disabled]="_disabled(row) || _disabledDueToLimit"
+      [value]="row"
+      [checked]="_isSelected"
+      (change)="_toggleRow($event)"
+      [aria-label]="ariaLabel"
+    ></dt-checkbox>
+  </dt-cell>`,
+  styleUrls: ['./selectors.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtTableRowSelector<T> implements OnDestroy {
+  /** The rows data used for selection */
+  @Input() row: T;
+
+  @Input() ariaLabel: '';
+
+  /**
+   * @internal Whether the checkbox should be disabled due to the selectionLimit
+   * Whether the checkbox is disabled is determined based on whether the predicate is true or
+   * the limit is reached and the row is currently not selected.
+   */
+  get _disabledDueToLimit(): boolean {
+    return this._selector.selectionLimitReached && !this._isSelected;
+  }
+
+  /** @internal Whether the row is currently selected */
+  get _isSelected(): boolean {
+    return this._selector.isSelected(this.row);
+  }
+
+  /** Subscription for selectionChanges stored for cleanup */
+  private _selectionChangedSubscription = Subscription.EMPTY;
+
+  constructor(
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _selector: DtTableSelection<T>,
+  ) {
+    this._selectionChangedSubscription = this._selector.selectionChange.subscribe(
+      () => {
+        this._changeDetectorRef.markForCheck();
+      },
+    );
+  }
+
+  ngOnDestroy(): void {
+    this._selectionChangedSubscription.unsubscribe();
+  }
+
+  /** @internal Callback when the row is toggled by interacting with the checkbox */
+  _toggleRow(changeEvent: DtCheckboxChange<T>): void {
+    if (changeEvent.checked) {
+      this._selector.select(this.row);
+    } else {
+      this._selector.deselect(this.row);
+    }
+  }
+
+  _disabled(row: T): boolean {
+    return this._selector.disabled(row);
+  }
+}
+
+/** Provides a master toggle selection in the table's header in combination with the DtTableSelection directive */
+@Component({
+  selector: 'dt-table-header-selector',
+  template: `
+    <dt-header-cell class="dt-selectable-cell">
+      <dt-checkbox
+        (change)="_toggleAllSelection($event)"
+        [checked]="_isAllSelected"
+        [indeterminate]="_isAnySelected"
+        [aria-label]="ariaLabel"
+      ></dt-checkbox>
+    </dt-header-cell>
+  `,
+  styleUrls: ['./selectors.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtTableHeaderSelector<T> implements OnDestroy {
+  @Input() ariaLabel: '';
```

Comment:
Same here, please rename the input to `aria-label` and do not assign a default value empty string.